### PR TITLE
Enable mouse scroll wheel control of sliders

### PR DESCRIPTION
**Pull request by stoneman**
This pull request adds the ability to manipulate the PID, filter, and motor sliders with the mouse's scroll wheel.

As demonstrated in the videos below, this is a convenient way of adjusting the sliders and, in the case of the motor sliders, enables you to easily set two motors to exactly the same value (which is almost impossible when clicking on them). The content pane can still be scrolled when the mouse is not directly over an active slider control.

https://github.com/betaflight/betaflight-configurator/assets/741710/bf1a58c9-7725-4384-b2ca-6280f1cfa7b2

https://github.com/betaflight/betaflight-configurator/assets/741710/5509de5b-c14f-4959-bd4d-0e1b943f415c

Since it is not a good experience having the sliders move about the screen while you are scrolling them, I have moved the PID/filter warning boxes from above the sliders to below them. This prevents the sliders from being pushed downwards when the warning appears.

I have also styled them to match the "notes" that they now sit on top of.

image

image

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by haslinghuis — APPROVED** [review]

**Comment by nerdCopter** [comment]
will this lead to accidental movements when simply wanting to page up/down.  maybe ctrl+wheel?

**Comment by chmelevskij** [comment]
> will this lead to accidental reconfiguration when simply wanting to page up/down. maybe ctrl+wheel?

Feels like this can be the case. Also, when you reduce the size of the window this could possibly cause issues with the scrolling as well.

My 2 cents on this around the UX. General, from experience, ppl get confused when scroll is hijacked by anything other than scroll.

IMO better UI/UX here would be to use `modifier` key like `shift` or `ctrl/cmd` to enable the scroll behavour instead.

File: src/js/tabs/motors.js
Comment (chmelevskij) [inline src/js/tabs/motors.js:1343]:
```suggestion
    const delta = e.originalEvent.deltaY > 0 ? -step : step;
```

File: src/js/tabs/pid_tuning.js
Comment (chmelevskij) [inline src/js/tabs/pid_tuning.js:2993]:
```suggestion
    const delta = e.originalEvent.deltaY > 0 ? -step : step;
```

**Comment by stoneman** [comment]
Since browsers use `shift` for horizontal scroll and `cmd/ctrl` for zoom, I've gone with `opt/alt`. Does that seem reasonable?

**Comment by nerdCopter** [comment]
* reasonable since `ctrl` activates zoom on web-app version.
* is here a way to activate when the mouse is anywhere in the `<TD>`? i.e. when the mouse is a few pixels above or below the slider, then vertical scrolling still happens.

**Comment by stoneman** [comment]
> is here a way to activate when the mouse is anywhere in the <TD>?

Yes, great idea - I have done so.

**Comment by stoneman** [comment]
Note that for `alt`+scroll wheel to work with the motor sliders, I've had to add `Alt` to the [list of keys](https://github.com/betaflight/betaflight-configurator/blob/4010fe3f8429c4fb64a179502e821dfaf31bf943/src/js/tabs/motors.js#L900) which do not [disable motor testing](https://github.com/betaflight/betaflight-configurator/blob/4010fe3f8429c4fb64a179502e821dfaf31bf943/src/js/tabs/motors.js#L942).

**Comment by nerdCopter** [comment]
> Note that for `alt/opt`+scroll wheel to work with the motor sliders, I've had to add `Alt` to the [list of keys](https://github.com/betaflight/betaflight-configurator/blob/4010fe3f8429c4fb64a179502e821dfaf31bf943/src/js/tabs/motors.js#L900) which do not [disable motor testing](https://github.com/betaflight/betaflight-configurator/blob/4010fe3f8429c4fb64a179502e821dfaf31bf943/src/js/tabs/motors.js#L942).

`fde5df0a`
* PID sliders work as expected :heart: 
* motors do not, i'm on Linux and ALT still disables

**Comment by stoneman** [comment]
> motors do not, i'm on Linux and ALT still disables

I've changed the ignored keys array / check to use numeric values which should solve this.

Having noticed the values of the existing `ignoredKeys`, I've discovered that the sliders can be controlled with arrow keys, page up/down, home, and end. Is this documented anywhere? If so, then I think I should add alt+scroll wheel to those docs.

**Comment by nerdCopter** [comment]
i dont know if it's required, but maybe rebase on master.  i'll approve this PR as well. :rocket:

**Comment by stoneman** [comment]
Could add a help tip like this...

image

image

...but adding that info to the help for some of the pid/filter sliders might be a bit much...

image

🤔

**Comment by stoneman** [comment]
rebased

**Comment by nerdCopter** [comment]
i don't think it needs to be verbalized in help-tips, personally.
can be mentioned on release notes maybe.

**Review by nerdCopter — APPROVED** [review]

**Comment by stoneman** [comment]
pushed a change to reduce code duplication and also to call `slider.trigger('change');` so that `setDirty(true)` is called. (I noticed that the profile/rate profile drop downs were getting disabled when clicking on a slider but not when scrolling them)

**Comment by stoneman** [comment]
Pushed another change to reduce duplication ...and now I see that the duplication check is complaining about `src/js/deubg.js`, which I haven't changed 🙈 Guess I should just ignore that check

File: src/js/tabs/motors.js
Comment (haslinghuis) [inline src/js/tabs/motors.js:925]: Wrote this function using `e.code` because 
https://medium.com/@danziger/both-e-which-and-e-keycode-ab9252580d08

File: src/js/tabs/motors.js
Comment (haslinghuis) [inline src/js/tabs/motors.js:907]: Why revert to numbers? Just add `altLeft` and `altRight` instead.

**Comment by stoneman** [comment]
@haslinghuis It was because the alt keys were [not being ignored on Linux](https://github.com/betaflight/betaflight-configurator/pull/3582#issuecomment-1729656448) but I guess that's because I used `Alt` when I should have used `AltLeft` and `AltRight`. I've reverted/amended the change.

**Comment by stoneman** [comment]
CI failed due to `error An unexpected error occurred: "https://registry.yarnpkg.com/abab/-/abab-2.0.6.tgz: Request failed \"502 Bad Gateway\"".` I don't see the "re-run job" option.

**Comment by nerdCopter** [comment]
`795c4aec` working on Linux

**Pull request merged**
